python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/utils.py

```diff
@@ -252,8 +278,14 @@ def _update_common_interface_cfg(
         self._update_interface_multicast_config(p2p_link, interface)
 
         if p2p_link.include_in_underlay_protocol:
-            if (self.inputs.underlay_rfc5549 and p2p_link.routing_protocol != "ebgp") or p2p_link.ipv6_enable is True:
+            if (
+                (self.inputs.underlay_rfc5549 and p2p_link.routing_protocol != "ebgp")
+                or p2p_link.ipv6_enable is True
+                or self.shared_utils.underlay_ipv6 is True
```

Comment:
This extra condition will be breaking, so we will need to write a porting guide entry, where in the diff we show they can set `ipv6_enable: false` to prevent the change.

Also since we now use strong typing we can rely on __bool__:

```suggestion
                or self.shared_utils.underlay_ipv6
```

Reply:
Have added porting guide entry. Let me know if it looks ok.